src/clusterfuzz/_internal/bot/fuzzers/afl/launcher.py

```diff
@@ -1469,6 +1470,42 @@ def prepare_runner(fuzzer_path,
   return runner
 
 
+def minimize_corpus(target_path, arguments, input_dirs, output_dir,
+                    reproducers_dir, max_time):
+  """Minimizes the corpora in |input_dirs| and writes the resulting corpus to
+  |output_dir|."""
+  del arguments
+  del reproducers_dir
+
+  set_additional_sanitizer_options_for_afl_fuzz()
+
+  # AFL minimizes corpora using the afl-cmin tool.
+  afl_tools_path = os.path.dirname(target_path)
+  afl_cmin_path = os.path.join(afl_tools_path, 'afl-cmin')
+  afl_cmin_runner = new_process.UnicodeProcessRunner(afl_cmin_path)
+
+  os.chmod(target_path, 0o775)
+  os.chmod(afl_cmin_path, 0o775)
+
+  # afl-cmin only works on one directory, copy all inputs to a single directory.
+  with tempfile.TemporaryDirectory as combined_input_dir:
+    for input_dir in input_dirs:
+      for input_file in list_full_file_paths(input_dir):
+        shutil.copyfile(
+            input_file,
+            os.path.join(combined_input_dir, os.path.basename(input_file)))
+
+    arguments = [
+        '-i', combined_input_dir, '-o', output_dir, '-m', 'none', '--',
+        target_path
+    ]
+    env = {'AFL_ALLOW_TMP': '1'}
+    max_time = max_time + AFL_CLEAN_EXIT_TIME
+    # What should we do about crashes?
```

Comment:
@vanhauser-thc What does AFL do with testcases that cause crashes during corpus minimization?

Reply:
Default: afl-cmin will nor minimize nor copy over crashes
`-C` option: afl-min will only minimize (and copy over) crashes

do you need a different behaviour?

Reply:
The behavior of libFuzzer (which I guess ClusterFuzz expects) is a combination of both, the unique noncrash testcases are copied to the corpus dir, but the crash testcases are copied to a different directory (and don't need to be deduplicated).

Reply:
I suppose worst case scenario we could just use the current default behavior for the cost of maybe findings bugs slower.